### DISCUSSION TITLE
Recommendations for a loss that has neighborhood awareness?

### DISCUSSION THREAD

**Discussion by JulienMaille** [isaaccorley/torchseg#27] [Q&A]
I'm training some model on classes that sometime are present in only a dozen pixels in an image (sometimes even less) and I get huge score differences from one extra or missing pixel.
Are their some loss (and score) function that better handle these case and won't penalize a false-positive pixel if it is near one that is present in the ground truth? I'm currently using micro-IoU.

**Comment by notprime** [comment]
Hey, probably you should look at generalized losses, like Generalized Dice Loss. Those losses use weights to handle classes differently, so that you can manage those situations.

You could also look at boundary losses, especially useful when dealing with objects that are small when compared to other classes or background.

We are currently working on an overalhaul of our losses. We will implement both generalized and boundary losses
